add navigation back to top-level stoprint (part 1)

drewvolz opened the pull request:
Adds stoprint top-level navigation back

* removes toplevelviewprops
* adds stoprint top-level routes back
* open settings button is broken here but fixed in the follow-up PR

drewvolz committed:
add navigation views and titles back to print release views

drewvolz committed:
remove toplevelviewprops from stoprint

drewvolz committed:
adding returns to functions that need them

drewvolz committed:
add navigation types to printer list and print job release

drewvolz committed:
refactor print jobs to use new navigation and functional calls

drewvolz committed:
refactor printers list to use new navigation and functional calls

drewvolz committed:
refactor print release to use new navigation and functional calls

drewvolz committed:
update problems, errors, and counts

drewvolz committed:
update counts, errors, problems

hawkrives commented on source/views/stoprint/print-jobs.tsx:
```diff
@@ -100,13 +94,13 @@ class PrintJobsView extends React.PureComponent<Props, State> {
 	keyExtractor = (item: PrintJob) => item.id.toString()
 
-	openSettings = () => this.props.navigation.navigate('SettingsView')
+	openSettings = () => navigation.navigate('SettingsView')
```
Where does `navigation` come from?

drewvolz replied:
I have a bullet point in the PR description but I should have made it a comment in the code.

> open settings button is broken here but fixed in the follow-up PR

I can revert this if you'd like, but I think the plan is to merge part 2 into part 1 before this lands.

drewvolz committed:
Merge branch 'drew/stoprint-part-2' of https://github.com/StoDevX/AAO-React-Native into drew/stoprint-part-2

# Conflicts:
#	eslint-problems
#	tsc-counts
#	tsc-errors

drewvolz committed:
replace any with void for updatePrinters

drewvolz committed:
fix dispatch not being called in updatePrinters

drewvolz committed:
mark error as possibly null instead of undefined

drewvolz committed:
add DebugNotice button to navigation buttons

a button to serve as a visual identifier of views serving up data that
is mocked

drewvolz committed:
add debug button to views showing mock stoprint data

drewvolz committed:
add mock stoprint endpoints and mock data

drewvolz committed:
use the mock data endpoints for stoprint in debug

drewvolz committed:
fix a stoprint type for success

drewvolz committed:
add mocked stoprint check to redux updatePrinters and updatePrintJobs

drewvolz committed:
add mocked stoprint checks when checking login status

drewvolz committed:
define printjobs error to be possibly undefined instead of null

drewvolz committed:
add username from login credentials instead of from navigation params

this is the same issue and fix that Hannes fix three years ago in a
separate PR that fixed releaseJob, whereas this fixes cancelJob
https://github.com/StoDevX/AAO-React-Native/pull/3730

we we were trying to extract the username from the navigation params
rather than loading them using loadLoginCredentials. I guess this never
got updated when we moved away from storing login credentials in redux
itself.

drewvolz committed:
add mock data loading to the heldJob function

* prevent login checking if we are serving up mocked data
* branch to fetch mock data from fetching production data
* depend upon printer object because properties may be undefined

drewvolz committed:
ensure that the printer object is not undefined in print release views

drewvolz committed:
update counts, problems, and errors

drewvolz committed:
remove unused and commented out functions

drewvolz committed:
add debug notice to mocked stoprint list of printers

drewvolz committed:
remove TS2532 (again) by setting print job error types to string or null

drewvolz committed:
update counts, problems, and errors

hawkrives reviewed (approved):
Rubber stamped - I don't have the mental bandwidth to review for a few weeks.

drewvolz committed:
turn off mocked stoprint setting

drewvolz committed:
Merge pull request #6078 from StoDevX/drew/stoprint-part-3

stoprint refactor –– mock data (part 3)

drewvolz committed:
Merge pull request #6074 from StoDevX/drew/stoprint-part-2

stoprint refactor (part 2)

drewvolz committed:
update type check on mocked stoprint

drewvolz committed:
update problems and errors

drewvolz committed:
Merge branch 'master' into drew/stoprint
